warehouse/query-core/src/main/java/datawave/query/planner/scanhints/PriorityScanHintRule.java

```diff
@@ -0,0 +1,19 @@
+package datawave.query.planner.scanhints;
+
+import org.apache.commons.jexl3.parser.JexlNode;
+
+import datawave.query.config.ScanHintRule;
+
+/**
+ * Scan hints that are used to set a priority in accumulo,
+ *
+ * @see <a href="https://accumulo.apache.org/docs/2.x/administration/scan-executors">https://accumulo.apache.org/docs/2.x/administration/scan-executors</a>
+ */
+public abstract class PriorityScanHintRule extends BaseScanHintRule<JexlNode> {
```

Comment:
What are we thinking of for an implementation of this one?

Reply:
I'd like to put together another PR that leverages the HintScanPrioritizer (from https://accumulo.apache.org/docs/2.x/administration/scan-executors) and has a weighted visitor to arrive at a priority based on the number of and type of query terms.